miner_exporter.py

```diff
@@ -63,379 +51,163 @@
-    c = line.split(',')
-    #print(f"{len(c)} {c}")
-    if len(c) == 10:
-      if c[0] == 'name' and c[1] == 'owner_address':
-        # header line
-        continue
-
-      (val_name,address,last_heartbeat,stake,status,version,tenure_penalty,dkg_penalty,performance_penalty,total_penalty) = c
-      if miner_name == val_name:
-        log.debug(f"have pen line: {c}")
-        tenure_penalty_val = try_float(tenure_penalty)
-        dkg_penalty_val = try_float(dkg_penalty)
-        performance_penalty_val = try_float(performance_penalty)
-        total_penalty_val = try_float(total_penalty)
-        least_heartbeat=try_float(last_heartbeat)
-
-        log.info(f"L penalty: {total_penalty_val}")
-        LEDGER_PENALTY.labels('ledger_penalties', 'tenure', miner_name).set(tenure_penalty_val)
-        LEDGER_PENALTY.labels('ledger_penalties', 'dkg', miner_name).set(dkg_penalty_val)
-        LEDGER_PENALTY.labels('ledger_penalties', 'performance', miner_name).set(performance_penalty_val)
-        LEDGER_PENALTY.labels('ledger_penalties', 'total', miner_name).set(total_penalty_val)
-        BLOCKAGE.labels('last_heartbeat', miner_name).set(last_heartbeat)
-
-    elif len(line) == 0:
-      # empty lines are fine
-      pass
-    else:
-      log.warning(f"failed to grok line: {c}; section count: {len(c)}")
-
-
-def collect_miner_version(docker_container, miner_name):
-  out = docker_container.exec_run('miner versions')
-  results = out.output.decode('utf-8').split("\n")
-  # sample output
-  # $ docker exec validator miner versions
-  # Installed versions:
-  # * 0.1.48	permanent
-  for line in results:
-    if m := re.match('^\*\s+([\d\.]+)(.*)', line):
-      miner_version = m.group(1)
-      log.info(f"found miner version: {miner_version}")
-      VALIDATOR_VERSION.labels(miner_name).info({'version': miner_version})
+    HBBFT_PERF.labels('hbbft_perf','Penalty', my_label).set(hval.get('pen_val', 0))
+    HBBFT_PERF.labels('hbbft_perf','BBA_Total', my_label).set(hval.get('bba_tot', 0))
+    HBBFT_PERF.labels('hbbft_perf','BBA_Votes', my_label).set(hval.get('bba_completions', 0))
+    HBBFT_PERF.labels('hbbft_perf','Seen_Total', my_label).set(hval.get('seen_tot', 0))
+    HBBFT_PERF.labels('hbbft_perf','Seen_Votes', my_label).set(hval.get('seen_votes', 0))
+    HBBFT_PERF.labels('hbbft_perf','BBA_Last', my_label).set(hval.get('bba_last_val', 0))
+    HBBFT_PERF.labels('hbbft_perf','Seen_Last', my_label).set(hval.get('seen_last_val', 0))
+    HBBFT_PERF.labels('hbbft_perf','Tenure', my_label).set(hval.get('tenure', 0))
+
+    if peer_book_info is not None:
+        connections = peer_book_info[0]['connection_count']
+        CONNECTIONS.labels('connections', my_label).set(connections)
+        sessions = len(peer_book_info[0]['sessions'])
+        SESSIONS.labels('sessions', my_label).set(sessions)
 
 
 if __name__ == '__main__':
   prometheus_client.start_http_server(9825) # 9-VAL on your phone
+  miner = MinerJSONRPC('http://localhost:4467/')
```

Comment:
Does localhost work for docker? I thought someone determined when JSONRPC came out that docker won't pick up that endpoint. Or did this PR fix this issues? https://github.com/helium/miner/pull/884